Repositories/ListingRepository.cs

```diff
@@ -27,6 +28,18 @@ public IEnumerable<Listing> GetAllListings()
             return JsonSerializer.Deserialize<List<Listing>>(jsonString);
         }
 
+        public IEnumerable<Listing> GetAllListings(string id)
+        {
+            List<Listing> listings = GetAllListings().ToList();
+            return listings.Where(a => a.OwnerId == id).ToList();
```

Comment:
.ToList() can be removed in this and probably all other methods of this class as we return IEnumerables.